crates/pallet-executor/src/lib.rs

```diff
@@ -412,6 +462,31 @@ impl<T: Config> Pallet<T> {
     ) -> Result<(), Error<T>> {
         Ok(())
     }
+
+    /// Prune old execution receipts up to (but not including) `up_to`.
+    fn prune_receipts_up_to(up_to: BlockNumber) {
+        ReceiptsRange::<T>::mutate(|range| {
```

Comment:
ERs are supposed to be sent in order, right? Why do we need to mess with ranges here at all, shouldn't everything before `up_to` be already pruned anyway?

Reply:
Consider the receipts pruning depth is changed.

Reply:
Then we can try to remove block `up_to - 1` and go down until we find storage item which didn't exist anyway. Or there could be a migration added if we go with runtime upgrade route, then we don't need to worry about it during regular execution at all.

Reply:
Well, I believe a migration will be necessary if we choose the runtime upgrade way, the work is still there, just when to do it. I don't have a strong opinion in the long run as mention above, runtime upgrade can save some limited state storage, but a sudo interface is more convenient and lightweight for changing the parameter.

Reply:
I just feel like we're having a storage item we need to keep up to date just for the sake of doing something that is only really needed IF we change pruning depth, which we can do either with runtime upgrade or, as have shown above, without it even. Just less storage items to worry about.

I'm fine doing it with sudo for now, but I'd prefer runtime upgrade path for sure. And regardless of that I think we should remove `ReceiptsRange` to make things simpler.

Reply:
Removing `ReceiptsRange` will fail the test set_new_receipts_pruning_depth_should_work, so I simply remove everything and resort to the runtime upgrade path, the receipts cache cleaning will have to be handled along with the runtime uprade. https://github.com/subspace/subspace/pull/427/commits/b55de80fb38cf89ddb1f64a50c9dc617d0f2a922